protocol.md

```diff
@@ -0,0 +1,196 @@
+
+## Content types
+
+Here is a list of supported content types:
+* `text/plain` identifies a message which content is a plain text
+
+TODO: are there other types? How is it useful? Sending Eth requests?
+
+## Message types
+
+Here is a list of supported message types:
+* `public-group-user-message` is a message to the public group
+* `user-message` is a private message.
+
+TODO: how is it useful? What are other supported message types?
+
+## Message ordering
+
+TBD
+
+## Quoting
+
+TBD
+
+# Whisper adapter
+
+Whisper has been chosen as an messages exchange protocol because it was designed as an off-chain communcation layer for the Ethereum nodes. It supports e2e encryption and uses epidemic spread to route data to all members of the network. It also provides [darkness to some extent](https://github.com/ethereum/go-ethereum/wiki/Achieving-Darkness).
+
+However, Whisper was not designed to handle huge number of messages and real-time communication. These are the tradeoffs that we accepted when implementing this protocol.
+
+This protocol can either work using a Whisper service which requires a protocol implementation to run in the same process as a Whisper node as well as using a Whisper client which might run as a separate Ethereum node and communicate through IPC or WebSocket.
+
+There is some tight coupling between the payload and Whisper:
+* Whisper message topic depends on the actual message type
+* Whisper message uses a different key (asymmetric or symmetric) depending on the actual message type
+
+## Whisper node configuration
+
+If you want to run a Whisper node and receive messages from Status clients, you need to configure it properly, otherwise, some message might not be received.
+
+Whisper's Proof Of Work algorithm is used to to deter denial of service and various spam/flood attacks against the Whisper network. The sender of a message must perform some work which in this case means performing some computations. Because Status main client is a mobile client, this easily leads to battery draining and poor performance of the app itself. Hence, these clients send envelopes with the following settings:
+* proof-of-work equal `0.002`
+* proof-of-work time equal to `10`
+* time-to-live equal `10`
+
+If you want to receive messages from a mobile client, you need to run Whisper node with PoW set to `0.002`. In case of `geth`, this option can be overriden with `-shh.pow=0.002` flag.
+
+## Keys management
+
+TBD
+
+## Topic
+
+There are two types of Whisper topics the protocol uses:
+* static topic for private chats
+* dynamic topic based on a chat name for public chats
+
+Static topic is always the same and its hex representation is `0xf8946aac`. It is also called a discovery topic. Having only one topic for all private chats has an advantage as it's very hard to figure out who talks to who. A drawback is that everyone receives everyone's messages but they can decrypt only these they have private keys for.
```

Comment:
call out discovery topic as a special term, e.g. Define Constant, possibly earlier in doc

similar to calling out specific encryption algos/curves used